js/pro/woo.js

```diff
@@ -0,0 +1,722 @@
+'use strict';
+
+// ----------------------------------------------------------------------------
+
+const wooRest = require ('../woo');
+const { ExchangeError, AuthenticationError } = require ('../base/errors');
+const { ArrayCacheByTimestamp, ArrayCacheBySymbolById, ArrayCache } = require ('./base/Cache');
+const Precise = require ('../base/Precise');
+
+// ----------------------------------------------------------------------------
+
+module.exports = class woo extends wooRest {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'has': {
+                'ws': true,
+                'watchBalance': false,
+                'watchMyTrades': false,
+                'watchOHLCV': true,
+                'watchOrderBook': true,
+                'watchOrders': true,
+                'watchTicker': true,
+                'watchTickers': true,
+                'watchTrades': true,
+            },
+            'urls': {
+                'api': {
+                    'ws': {
+                        'public': 'wss://wss.woo.org/ws/stream',
+                        'private': 'wss://wss.woo.network/v2/ws/private/stream',
+                    },
+                },
+                'test': {
+                    'ws': {
+                        'public': 'wss://wss.staging.woo.org/ws/stream',
+                        'private': 'wss://wss.staging.woo.org/v2/ws/private/stream',
+                    },
+                },
+            },
+            'requiredCredentials': {
+                'apiKey': false,
+                'secret': false,
+                'uid': true,
+            },
+            'options': {
+                'tradesLimit': 1000,
+                'ordersLimit': 1000,
+                'requestId': {},
+            },
+            'streaming': {
+                'ping': this.ping,
+                'keepAlive': 10000,
+            },
+        });
+    }
+
+    requestId (url) {
+        const options = this.safeValue (this.options, 'requestId', {});
```

Comment:
if the connection breaks and the user re-connects, shouldn't we reset this counter?

Example:
```Javacript
    while(true) {
        try{
            trades = await exchange.watchTrades('BTC/USDT');
        } catch(e) {
            // do nothing
        }
    }
```

Reply:
actually the requestId doesn't matter, just needs to be unique afaik